src/apps/watchfaces/OswAppWatchfaceFitnessAnalog.cpp

```diff
@@ -0,0 +1,323 @@
+    this->knownButtonStates[Button::BUTTON_DOWN] = (OswAppV2::ButtonStateNames) (this->knownButtonStates[Button::BUTTON_DOWN] | OswAppV2::ButtonStateNames::DOUBLE_PRESS); // OR to set the bit
+
+    this->lastTime = time(nullptr); // use
+    printf("xxx onStart\n");
+}
+
+void OswAppWatchfaceFitnessAnalog::onLoop() {
+    OswAppV2::onLoop();
+
+    this->needsRedraw = this->needsRedraw or time(nullptr) != this->lastTime; // redraw every second
+}
+
+
+void OswAppWatchfaceFitnessAnalog::onDraw() {
+    OswHal* hal = OswHal::getInstance();
+    hal->setCPUClock(OSW_PLATFORM_DEFAULT_CPUFREQ);
+
+    const int iter = 1;
+    #ifndef OSW_EMULATOR
+        unsigned long old_micros = micros();
+        for (int i=iter; i>0; --i)
+            test();
+        printf("xxxx time for onDraws (average over %d iterations) %f ms.\n", iter, (micros()-old_micros)/(float) iter/1000);
+        Serial.flush();
+    #else
+        unsigned long old_millis = millis();
+        for (int i=iter; i>0; --i)
+            test();
+        printf("xxxx time for onDraws (average over %d iterations) %f ms.\n", iter, (millis()-old_millis)/(float) iter);
+    #endif
+    hal->setCPUClock(20);
+}
+
+void OswAppWatchfaceFitnessAnalog::test() {
+    OswAppV2::onDraw();
+
+    OswHal* hal = OswHal::getInstance();
+
+    uint32_t second = 0;
+    uint32_t minute = 0;
+    uint32_t hour = 0;
+    bool afterNoon;
+    hal->getLocalTime(&hour, &minute, &second, &afterNoon);
+
+    if (this->screen == 0) {
+        #if OSW_PLATFORM_ENVIRONMENT_ACCELEROMETER == 1
+            showFitnessTracking(hal);
+        #endif
+
+        drawWatchFace(hal, hour, minute, second, afterNoon);
+    } else if (this->screen == 1) {
+        drawDateFace(hal, hour, minute, second, afterNoon);
+
+        static int wait_time = 1;
+        if (wait_time >= 0)
+            --wait_time;
+        else {
+            this->screen = 0;
+            wait_time = 1;
+        }
+    }
+        
+    this->lastTime = time(nullptr);
+}
+
+void OswAppWatchfaceFitnessAnalog::onButton(Button id, bool up, OswAppV2::ButtonStateNames state) {
+    OswAppV2::onButton(id, up, state);
+
+    if(!up) return;
+
+    this->lastTime = time(nullptr); // reset the timeout, as the user interacted with the device
+
+    if (state == OswAppV2::ButtonStateNames::DOUBLE_PRESS) {
+        if (this->screen < 1)
+            ++this->screen;
+        return;       
+    }
+
+    if(OswAppWatchface::onButtonDefaults(*this, id, up, state))
+        return; // if the button was handled by the defaults, we are done here
+}
+
+/**
+ * Fast RGB565 pixel blending
+ * @param fg      The foreground color in uint16_t RGB565 format
+ * @param bg      The background color in uint16_t RGB565 format
+ * @param alpha   The alpha in range 0-255
+ **/
+uint16_t alphaBlendRGB565( uint32_t fg, uint32_t bg, uint8_t alpha ){
```

Comment:
:heart_eyes: I was always too lazy for that - but I also recently learned that the CPU is fast enough for that stuff - thank you!